web/tests/notifications.test.js

```diff
@@ -119,7 +126,31 @@ test("message_is_notifiable", () => {
     assert.equal(notifications.should_send_audible_notification(message), true);
     assert.equal(notifications.message_is_notifiable(message), true);
 
-    // Case 4:
+    // Case 4: If the message has been sent to a followed topic,
+    // DO visually notify the user if 'enable_followed_topic_desktop_notifications'
+    // is enabled.
+    // Messages to followed topics trumps muting
+    message = {
+        id: 30,
+        content: "message number 3",
+        sent_by_me: false,
+        notification_sent: false,
+        mentioned: false,
+        mentioned_me_directly: false,
+        type: "stream",
+        stream: "general",
+        stream_id: general.stream_id,
+        topic: "followed topic",
+    };
+    assert.equal(notifications.should_send_desktop_notification(message), true);
+    assert.equal(notifications.message_is_notifiable(message), true);
```

Comment:
We should also verify here that for normal topics, `message_is_notifiable == false`

Reply:
I think this test already covers all the cases where `message_is_notifiable` should be `false`.

For normal topics (not muted and no visibility policy set), `message_is_notifiable` will be `true`, and should_send desktop/audible notifications will be determined by the respective settings. (This is on line 285)